src/main/scala/com/microsoft/hyperspace/telemetry/HyperspaceEvent.scala

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.telemetry
+
+import com.microsoft.hyperspace.index.IndexLogEntry
+
+/**
+ * Class for common app info.
+ *
+ * @param sparkUser Spark user.
+ * @param appId Spark Application Id.
+ * @param appName Spark App Name.
+ */
+case class AppInfo(sparkUser: String, appId: String, appName: String)
+
+/**
+ * Trait for all Hyperspace events.
+ */
+trait HyperspaceEvent
+
+/**
+ * General index CRUD event.
+ *
+ * @param appInfo AppInfo for spark application.* @param index Related indexes.
+ * @param message Message about event.
+ */
+class HyperspaceIndexCRUDEvent(appInfo: AppInfo, index: IndexLogEntry, message: String)
```

Comment:
looks like you are using this only for grouping anyway? then just make this as trait and remove fields? I see you have fields in case class and here so you are duplicating? (not quite sure about your intention here)

Reply:
Thanks, fixed